psycop/common/feature_generation/data_checks/flattened/feature_describer_tsflattener_v2.py

```diff
@@ -0,0 +1,98 @@
+    """Get proportion of series that is equal to the value argument."""
+    if value == "nan":
+        return round(series.is_nan().mean(), 2)  # type: ignore
+    return round(series.eq(float(value)).mean(), 2)  # type: ignore
+
+
+def parse_predictor_column_name(
+    col_name: str,
+    is_static: Callable[[str], bool] = tsflattener_v2_column_is_static,
+    feature_name_regex: str = r"[a-zA-Z]+_(.*?)_(?=fallback|within)",
+    fallback_regex: str = r"_fallback_(.+)$",
+    time_interval_start_regex: str = r"_within_([0-9]+)",
+    time_interval_end_regex: str = "_to_([0-9]+)_",
+    time_interval_format_regex: str = r"to_[0-9]+_([a-z]+)_",
+    resolve_multiple_strategy_regex: str = r"([a-z]+)_fallback",
+) -> ParsedPredictorColumn:
+    col_is_static = is_static(col_name)
+    feature_name = re.search(feature_name_regex, col_name).group(1)  # type: ignore
+    fallback = re.search(fallback_regex, col_name).group(1)  # type: ignore
+    time_interval_start = (
+        re.search(time_interval_start_regex, col_name).group(1) if not col_is_static else "N/A"  # type: ignore
+    )
+    time_interval_end = (
+        re.search(time_interval_end_regex, col_name).group(1) if not col_is_static else "N/A"  # type: ignore
+    )
+    time_interval_format = (
+        re.search(time_interval_format_regex, col_name).group(1) if not col_is_static else "N/A"  # type: ignore
+    )
+    resolve_multiple_strategy = (
+        re.search(resolve_multiple_strategy_regex, col_name).group(1)  # type: ignore
+        if not col_is_static
+        else "N/A"
+    )
+    return ParsedPredictorColumn(
+        col_name=col_name,
+        feature_name=feature_name,
+        fallback=fallback,
+        time_interval_start=time_interval_start,
+        time_interval_end=time_interval_end,
+        time_interval_format=time_interval_format,
+        resolve_multiple_strategy=resolve_multiple_strategy,
+        is_static=is_static(col_name),
+    )
+
+
+def generate_feature_description_df(
+    df: pl.DataFrame,
+    column_name_parser: Callable[[str], ParsedPredictorColumn] = parse_predictor_column_name,
+) -> pl.DataFrame:
+    feature_rows = []
+    for col in df.columns:
+        parsed_col = column_name_parser(col)
+        n_unique = df[col].n_unique()
+        proportion_missing = round(df[col].is_nan().mean(), 2)  # type: ignore
+        mean = round(df[col].drop_nans().mean(), 2)  # type: ignore
+        proportion_using_fallback = get_value_proportion(df[col], parsed_col.fallback)
+
+        feature_description = {
+            "Feature name": parsed_col.feature_name,
+            "Lookbehind period": parsed_col.time_interval_start
+            if parsed_col.is_static
+            else f"{parsed_col.time_interval_start} to {parsed_col.time_interval_end} {parsed_col.time_interval_format}",
+            "Resolve multiple": parsed_col.resolve_multiple_strategy,
+            "Fallback strategy": parsed_col.fallback,
```

Comment:
maybe a bit off topic but i just wondered whether "fallback strategy" is something we came up with or a more general naming convention? i would tend to think of this "imputation value"

Reply:
We came up with fallback strategy. I agree that this is basically imputation, but perhaps it might be conflated with the other type of imputation that is (optionally) done during preprocessing? We use fallback in tsflattener, so might be good to be consistent?

Reply:
Also, some of the fallbacks are not estimated from the data (e.g. the NaN), so they are technically not imputed until preprocessing.